pkg/cmd/fixcerts/fixcerts.go

```diff
@@ -0,0 +1,431 @@
+	}
+
+	definitions := []struct {
+		objectType  string
+		name        string
+		namespace   string
+		toplevelDir string
+	}{
+		// Fix kube-apiserver serving certs
+		{
+			objectType:  secretsType,
+			name:        "service-network-serving-certkey",
+			namespace:   kubeapiserveroperatorclient.TargetNamespace,
+			toplevelDir: *kubeApiserverCertsDir,
+		},
+		{
+			objectType:  configmapsType,
+			name:        "client-ca",
+			namespace:   kubeapiserveroperatorclient.TargetNamespace,
+			toplevelDir: *kubeApiserverCertsDir,
+		},
+		{
+			objectType:  secretsType,
+			name:        "localhost-serving-cert-certkey",
+			namespace:   kubeapiserveroperatorclient.TargetNamespace,
+			toplevelDir: *kubeApiserverCertsDir,
+		},
+		{
+			objectType:  secretsType,
+			name:        "internal-loadbalancer-serving-certkey",
+			namespace:   kubeapiserveroperatorclient.TargetNamespace,
+			toplevelDir: *kubeApiserverCertsDir,
+		},
+
+		// Fix controller-manager certs
+		{
+			objectType:  secretsType,
+			name:        "kube-controller-manager-client-cert-key",
+			namespace:   kubeapiserveroperatorclient.GlobalMachineSpecifiedConfigNamespace,
+			toplevelDir: *kubeControllerManagerResourceDir,
+		},
+		{
+			objectType:  secretsType,
+			name:        "csr-signer",
+			namespace:   kubecontrollermanageroperatorclient.OperatorNamespace,
+			toplevelDir: *kubeControllerManagerResourceDir,
+		},
+
+		// Fix scheduler certs
+		{
+			objectType:  secretsType,
+			name:        "kube-scheduler-client-cert-key",
+			namespace:   kubeapiserveroperatorclient.GlobalMachineSpecifiedConfigNamespace,
+			toplevelDir: *kubeSchedulerResourceDir,
+		},
+	}
+	for _, def := range definitions {
+		data := map[string][]byte{}
+		switch def.objectType {
+		case secretsType:
+			klog.V(2).Infof("Getting secret '%s/%s'", def.namespace, def.name)
+			secret, err := kubeClient.CoreV1().Secrets(def.namespace).Get(def.name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			data = secret.Data
+
+		case configmapsType:
+			klog.V(2).Infof("Getting configmap '%s/%s'", def.namespace, def.name)
+			configMap, err := kubeClient.CoreV1().ConfigMaps(def.namespace).Get(def.name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			for k, v := range configMap.Data {
+				data[k] = []byte(v)
+			}
+		default:
+			return fmt.Errorf("unknown object type %q", def.objectType)
+		}
+
+		dir := filepath.Join(def.toplevelDir, def.objectType, def.name)
+		err = os.MkdirAll(dir, 0755)
+		if err != nil {
+			return fmt.Errorf("failed to create dir %q: %v", dir, err)
+		}
+
+		for name, bytes := range data {
+			filePath := filepath.Join(dir, name)
+			err := recovery.EnsureFileContent(filePath, bytes)
+			if err != nil {
+				return fmt.Errorf("failed to write file %q: %v", filePath, err)
+			}
+		}
+	}
+
+	timestamp := time.Now().Format(time.RFC3339)
+	kubeControllerManagerPod.Annotations["force-triggered-by-fix-certs-at"] = timestamp
```

Comment:
I expected to see the kube-apiserver pod here too.  Let's do them all, just to be sure.